ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -69,5 +82,100 @@ def authenticate(self, request, username=None, password=None, **kwargs):
                     "is_superuser": user.is_superuser,
                 },
             )
-
         return update_user_claims(user, self.database_instance, [])
+
+    def is_a_partially_migrated_controller_user(self, username, password):
+        """
+        Check if a user exists in the AuthenticatorUser table with the local authenticator provider.
+        If the user is valid, update the gateway users credentials with the controller credentials.
+        """
+        if not self.gateway_user_can_be_migrated(username):
+            logger.warning(f"User '{username}' cannot be migrated to the gateway.")
+            return False
+
+        if controller_user := self._get_controller_user(username, password):
+            # Validate controller_user is a dictionary
+            if not isinstance(controller_user, dict):
+                logger.warning(f"Unable to authenticate user '{username}' with controller. Invalid result format.")
+                return False
+
+            # Validate controller_user has a ldap_dn field, if it is not None, then the user is a local user
+            ldap_dn = controller_user.get("ldap_dn")
+            if ldap_dn is None or ldap_dn != "":
+                logger.warning(f"User '{username}' is unable to be confirmed as a local user.")
```

Comment:
I think we also have to check for `password == "$encrypted$"` too to ensure they are local. The social auth provides will make the user serializer remove the password field.

Reply:
Updated to include this check